compiler/hie-core/src/Development/IDE/LSP/CodeAction.hs

```diff
@@ -77,19 +90,68 @@ suggestAction contents Diagnostic{_range=_range@Range{..},..}
 
 suggestAction _ _ = []
 
+mkRenameEdit :: Maybe T.Text -> Range -> T.Text -> TextEdit
+mkRenameEdit contents range name =
+    if fromMaybe False maybeIsInfixFunction
+      then TextEdit range ("`" <> name <> "`")
+      else TextEdit range name
+  where
+    maybeIsInfixFunction = do
+      curr <- textInRange range <$> contents
+      pure $ "`" `T.isPrefixOf` curr && "`" `T.isSuffixOf` curr
+
+
+extractRenamableTerms :: T.Text -> [T.Text]
+extractRenamableTerms msg
+  -- Account for both "Variable not in scope" and "Not in scope"
+  | "ot in scope:" `T.isInfixOf` msg = extractSuggestions msg
+  | otherwise = []
+  where
+    extractSuggestions = map getEnclosed
+                       . concatMap singleSuggestions
+                       . filter isKnownSymbol
+                       . T.lines
+    singleSuggestions = T.splitOn "), " -- Each suggestion is comma delimited
+    isKnownSymbol t = " (imported from" `T.isInfixOf` t  || " (line " `T.isInfixOf` t
+    getEnclosed = T.dropWhile (== '‘')
```

Comment:
This isn’t quite right. GHC doesn’t always emit smart quotes. Instead there is a `useUnicode` flag in `DFlags` that controls this, see https://gitlab.haskell.org/ghc/ghc/blob/d0b45ac6984f245bce9de7ffcc7dad4a0046d344/compiler/utils/Outputable.hs#L609. That said, if it’s annoying to get access to this flag here I’m also fine with not supporting this for now. People should really be using unicode locales  at this point :slightly_smiling_face:

Reply:
Not fixing this for now since it seems a bit tricky to get right. (You have to be careful not to strip the `'` from `foldl'` for example.)

Reply:
Looks like the non-smart quotes are the default on Windows, at least for me. @ndmitchell what do you see?

Regardless, I suggest getting this merged now and one of us can try to fix it afterwards.